pkg/signature/kms/aws/signer.go

```diff
@@ -117,7 +117,8 @@ func (a *SignerVerifier) PublicKey(opts ...signature.PublicKeyOption) (crypto.Pu
 		opt.ApplyContext(&ctx)
 	}
 
-	return a.client.public(ctx)
+	cmk, err := a.client.getCMK(ctx)
```

Comment:
We should check if err != nil before accessing `cmk`, otherwise `cmk` might be nil